### PR TITLE
Description of VersionedCollapsingMergeTree

### PR DESCRIPTION
**Pull request by BayoNet**
I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=en

### CODE REVIEW

File: docs/en/operations/table_engines/versionedcollapsingmergetree.md [hunk 1]
```diff
@@ -0,0 +1,226 @@
+<a name="table_engine-versionedcollapsingmergetree"></a>
+
+# VersionedCollapsingMergeTree
+
+The engine inherits from [MergeTree](mergetree.md#table_engines-mergetree) and adds the logic of rows collapsing to data parts merge algorithm.
+
+`VersionedCollapsingMergeTree` asynchronously deletes (collapses) pairs of rows if all of the fields in a row are equivalent excepting the particular field `Sign` which can have `1` and `-1` values. Rows without a pair are kept. For more details see the [Collapsing](#versionedcollapsingmergetree-collapsing) section of the document.
```
Comment (alexey-milovidov) [inline docs/en/operations/table_engines/versionedcollapsingmergetree.md]: `if all of the fields in a row are equivalent`
Not true.

File: docs/en/operations/table_engines/versionedcollapsingmergetree.md [hunk 2]
```diff
@@ -0,0 +1,226 @@
+<a name="table_engine-versionedcollapsingmergetree"></a>
+
+# VersionedCollapsingMergeTree
```
Comment (alexey-milovidov) [inline docs/en/operations/table_engines/versionedcollapsingmergetree.md]: Cannot figure out the motivation for `VersionedCollapsingMergeTree` from this text.

File: docs/en/operations/table_engines/versionedcollapsingmergetree.md [hunk 3]
```diff
@@ -0,0 +1,229 @@
...
+
+The engine:
```
Comment (blinkov) [inline docs/en/operations/table_engines/versionedcollapsingmergetree.md]: "This engine" would sound better I think

File: docs/en/operations/table_engines/versionedcollapsingmergetree.md [hunk 4]
```diff
@@ -0,0 +1,229 @@
...
+2. Long growing arrays in columns reduce the efficiency of the engine due to load for writing. The more straightforward data, the higher efficiency.
+3. `SELECT` results depend strongly on the consistency of object changes history. Be accurate when preparing data for inserting. You can get unpredictable results in inconsistent data, for example, negative values for non-negative metrics such as session depth.
+
+### Algorithm
+
+When ClickHouse merges data parts, it deletes each pair of rows having the same primary key and version and different `Sign`. The order of rows does not matter.
+
+When ClickHouse merges data parts, it orders rows by the primary key. If the `Version` column is not in the primary key, ClickHouse adds it to the primary key implicitly as the last field and use for ordering.
+
+## Selecting data
```
Comment (blinkov) [inline docs/en/operations/table_engines/versionedcollapsingmergetree.md]: **D**ata

File: docs/en/operations/table_engines/versionedcollapsingmergetree.md [hunk 5]
```diff
@@ -0,0 +1,229 @@
...
+
+To finalize collapsing write a query with `GROUP BY` clause and aggregate functions that account for the sign. For example, to calculate quantity, use `sum(Sign)` instead of `count()`. To calculate the sum of something, use `sum(Sign * x)` instead of `sum(x)`, and so on, and also add `HAVING sum(Sign) > 0`.
+
+The aggregates `count`, `sum` and `avg` could be calculated this way. The aggregate `uniq` could be calculated if an object has at list one state not collapsed. The aggregates `min` and `max` could not be calculated because `VersionedCollapsingMergeTree` does not save values history of the collapsed states.
+
+If you need to extract the data with "collapsing" but without of aggregation (for example, to check whether rows are present whose newest values match certain conditions), you can use the `FINAL` modifier for the `FROM` clause. This approach is inefficient and should not be used with big tables.
+
+## Example of use
```
Comment (blinkov) [inline docs/en/operations/table_engines/versionedcollapsingmergetree.md]: **U**se

File: docs/en/operations/table_engines/versionedcollapsingmergetree.md [hunk 6]
```diff
@@ -0,0 +1,229 @@
+<a name="table_engine-versionedcollapsingmergetree"></a>
+
+# VersionedCollapsingMergeTree
+
+The engine:
+
+- Allows quick writing of continually changing states of objects.
```
Comment (blinkov) [inline docs/en/operations/table_engines/versionedcollapsingmergetree.md]: It'd be useful to briefly mention Version right here as a mechanism to restore order of writes coming from concurrent writers.

**Review by blinkov — CHANGES_REQUESTED** [review]

**Pull request merged**